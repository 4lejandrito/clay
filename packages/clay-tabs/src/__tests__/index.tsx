/**
 * Â© 2019 Liferay, Inc. <https://liferay.com>
 *
 * SPDX-License-Identifier: BSD-3-Clause
 */

import '@testing-library/jest-dom/extend-expect';
import ClayTabs from '..';
import React from 'react';
import {cleanup, fireEvent, render} from '@testing-library/react';

const ClayTabsWithItems = () => {
	const [activeTabKeyValue, setActiveTabKeyValue] = React.useState<number>(0);

	return (
		<>
			<ClayTabs>
				<ClayTabs.Item
					active={activeTabKeyValue == 0}
					onClick={() => setActiveTabKeyValue(0)}
				>
					{'Dummy1'}
				</ClayTabs.Item>
				<ClayTabs.Item
					active={activeTabKeyValue == 1}
					data-testid="tabItem2"
					onClick={() => setActiveTabKeyValue(1)}
				>
					{'Dummy2'}
				</ClayTabs.Item>
				<ClayTabs.Item
					active={activeTabKeyValue == 2}
					onClick={() => setActiveTabKeyValue(2)}
				>
					{'Dummy3'}
				</ClayTabs.Item>
			</ClayTabs>
			<ClayTabs.Content activeTabKey={activeTabKeyValue}>
				<ClayTabs.TabPane data-testid="tabPane1">
					{'Tab Content 1'}
				</ClayTabs.TabPane>
				<ClayTabs.TabPane data-testid="tabPane2">
					{'Tab Content 2'}
				</ClayTabs.TabPane>
				<ClayTabs.TabPane>{'Tab Content 3'}</ClayTabs.TabPane>
			</ClayTabs.Content>
		</>
	);
};

describe('ClayTabs', () => {
	afterEach(cleanup);

	it('renders', () => {
		const {container} = render(<ClayTabs />);

		expect(container).toMatchSnapshot();
	});

	it('renders justifying nav items', () => {
		const {container} = render(<ClayTabs justified />);

		expect(container).toMatchSnapshot();
	});

	it('renders with modern style', () => {
		const {container} = render(<ClayTabs modern />);

		expect(container).toMatchSnapshot();
	});

	it('renders with items', () => {
		const {container} = render(<ClayTabsWithItems />);

		expect(container).toMatchSnapshot();
	});

	it('renders nav items as anchors', () => {
		const {container} = render(
			<>
				<ClayTabs>
					<ClayTabs.Item active component="a" />
					{'One'}
					<ClayTabs.Item component="a">{'Two'}</ClayTabs.Item>
				</ClayTabs>
				<ClayTabs.Content activeTabKey={1}>
					<ClayTabs.TabPane>{'Content One'}</ClayTabs.TabPane>
					<ClayTabs.TabPane>{'Content Two'}</ClayTabs.TabPane>
				</ClayTabs.Content>
			</>
		);

		expect(container).toMatchSnapshot();
	});

	it('renders disabled nav items', () => {
		const {container} = render(
			<>
				<ClayTabs>
					<ClayTabs.Item active>{'One'}</ClayTabs.Item>
					<ClayTabs.Item>{'Two'}</ClayTabs.Item>
					<ClayTabs.Item disabled>{'Three'}</ClayTabs.Item>
				</ClayTabs>
				<ClayTabs.Content activeTabKey={1}>
					<ClayTabs.TabPane>{'Content One'}</ClayTabs.TabPane>
					<ClayTabs.TabPane>{'Content Two'}</ClayTabs.TabPane>
					<ClayTabs.TabPane>{'Content Three'}</ClayTabs.TabPane>
				</ClayTabs.Content>
			</>
		);

		expect(container).toMatchSnapshot();
	});

	it('emits a number when clicking an item', () => {
		const onClick = jest.fn();

		const {getAllByTestId} = render(
			<>
				<ClayTabs>
					<ClayTabs.Item active onClick={onClick}>
						{'One'}
					</ClayTabs.Item>
					<ClayTabs.Item component="a" onClick={onClick}>
						{'Two'}
					</ClayTabs.Item>
				</ClayTabs>
				<ClayTabs.Content activeTabKey={1}>
					<ClayTabs.TabPane>{'Content One'}</ClayTabs.TabPane>
					<ClayTabs.TabPane>{'Content Two'}</ClayTabs.TabPane>
				</ClayTabs.Content>
			</>
		);

		const tabItems = getAllByTestId('tabItem');

		fireEvent.click(tabItems[0]);
		expect(onClick).toBeCalled();

		fireEvent.click(tabItems[1]);
		expect(onClick).toBeCalled();
	});
});
