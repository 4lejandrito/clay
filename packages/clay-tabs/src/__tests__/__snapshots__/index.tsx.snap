// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ClayTabs renders 1`] = `
<div>
  <ul
    class="nav nav-underline"
    role="tablist"
  />
</div>
`;

exports[`ClayTabs renders disabled nav items 1`] = `
<div>
  <ul
    class="nav nav-underline"
    role="tablist"
  >
    <li
      class="nav-item"
    >
      <button
        aria-disabled="false"
        aria-selected="true"
        class="nav-link active btn btn-unstyled"
        data-testid="tabItem"
        role="tab"
        type="button"
      >
        One
      </button>
    </li>
    <li
      class="nav-item"
    >
      <button
        aria-disabled="true"
        aria-selected="false"
        class="nav-link btn btn-unstyled"
        data-testid="tabItem"
        role="tab"
        type="button"
      >
        Two
      </button>
    </li>
    <li
      class="nav-item"
    >
      <button
        aria-disabled="true"
        aria-selected="false"
        class="nav-link btn btn-unstyled disabled"
        data-testid="tabItem"
        role="tab"
        tabindex="-1"
        type="button"
      >
        Three
      </button>
    </li>
  </ul>
  <div
    class="tab-content"
  >
    <div
      class="tab-pane"
      role="tabpanel"
    >
      Content One
    </div>
    <div
      class="tab-pane active"
      role="tabpanel"
    >
      Content Two
    </div>
    <div
      class="tab-pane"
      role="tabpanel"
    >
      Content Three
    </div>
  </div>
</div>
`;

exports[`ClayTabs renders justifying nav items 1`] = `
<div>
  <ul
    class="nav nav-justified nav-underline"
    role="tablist"
  />
</div>
`;

exports[`ClayTabs renders nav items as anchors 1`] = `
<div>
  <ul
    class="nav nav-underline"
    role="tablist"
  >
    <li
      class="nav-item"
    >
      <a
        aria-disabled="false"
        aria-selected="true"
        class="nav-link active"
        data-testid="tabItem"
        role="tab"
      />
    </li>
    One
    <li
      class="nav-item"
    >
      <a
        aria-disabled="true"
        aria-selected="false"
        class="nav-link"
        data-testid="tabItem"
        role="tab"
      >
        Two
      </a>
    </li>
  </ul>
  <div
    class="tab-content"
  >
    <div
      class="tab-pane"
      role="tabpanel"
    >
      Content One
    </div>
    <div
      class="tab-pane active"
      role="tabpanel"
    >
      Content Two
    </div>
  </div>
</div>
`;

exports[`ClayTabs renders with items 1`] = `
<div>
  <ul
    class="nav nav-underline"
    role="tablist"
  >
    <li
      class="nav-item"
    >
      <button
        aria-disabled="false"
        aria-selected="true"
        class="nav-link active btn btn-unstyled"
        data-testid="tabItem"
        role="tab"
        type="button"
      >
        Dummy1
      </button>
    </li>
    <li
      class="nav-item"
      data-testid="tabItem2"
    >
      <button
        aria-disabled="true"
        aria-selected="false"
        class="nav-link btn btn-unstyled"
        data-testid="tabItem"
        role="tab"
        type="button"
      >
        Dummy2
      </button>
    </li>
    <li
      class="nav-item"
    >
      <button
        aria-disabled="true"
        aria-selected="false"
        class="nav-link btn btn-unstyled"
        data-testid="tabItem"
        role="tab"
        type="button"
      >
        Dummy3
      </button>
    </li>
  </ul>
  <div
    class="tab-content"
  >
    <div
      class="tab-pane active"
      data-testid="tabPane1"
      role="tabpanel"
    >
      Tab Content 1
    </div>
    <div
      class="tab-pane"
      data-testid="tabPane2"
      role="tabpanel"
    >
      Tab Content 2
    </div>
    <div
      class="tab-pane"
      role="tabpanel"
    >
      Tab Content 3
    </div>
  </div>
</div>
`;

exports[`ClayTabs renders with modern style 1`] = `
<div>
  <ul
    class="nav nav-underline"
    role="tablist"
  />
</div>
`;
